Utils/src/PDFWeightProducer.cc

```diff
@@ -192,9 +192,11 @@ void PDFWeightProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::Ev
 
       int id1 = genHandle->pdf()->id.first;
       double x1 = genHandle->pdf()->x.first;
+      id1 = std::abs(id1) == 21 ? 0 : id1;
```

Comment:
can you make one function that does this, so the pattern doesn't have to be repeated everywhere?

Reply:
Done